src/coreclr/src/jit/importer.cpp

```diff
@@ -4500,6 +4493,91 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                     result = NI_System_MathF_Round;
                 }
             }
+
+            if (strcmp(methodName, "SinInternal") == 0)
```

Comment:
This should be recognizing the public non-internal methods entrypoint, not the internal versions.